src/category_theory/limits/shapes/zero.lean

```diff
@@ -45,50 +47,55 @@ attribute [simp] has_zero_morphisms.comp_zero
 restate_axiom has_zero_morphisms.zero_comp'
 attribute [simp, reassoc] has_zero_morphisms.zero_comp
 
+section
+omit 𝒞
+
+instance has_zero_morphisms_pempty : has_zero_morphisms.{v} pempty.{v+1} :=
+{ has_zero := by tidy }
+
+instance has_zero_morphisms_punit : has_zero_morphisms.{v} punit.{v+1} :=
+{ has_zero := λ X Y, { zero := punit.star, } }
+end
+
 namespace has_zero_morphisms
 variables {C}
 
-/-- This lemma will be immediately superseded by `ext`, below. -/
```

Comment:
This still looks like it reverts a bunch of changes from #2180.

Reply:
Ugh. I really don't know how I messed up that merge so badly. I think it's fixed now.